### PR TITLE
Update docs to point to minor version branches

### PR DESCRIPTION
**Pull request by marcusdacoregio**
Closes gh-9986

### CODE REVIEW

File: docs/manual/spring-security-docs-manual.gradle [hunk 1]
```diff
@@ -16,9 +16,10 @@ asciidoctor {
 asciidoctorj {
 	def docsTag = snapshotBuild ? 'current' : project.version
 	def ghTag = snapshotBuild ? 'main' : project.version
+	def samplesBranch = computeSamplesBranch(project.version as String)
```
Comment (jzheaux) [inline docs/manual/spring-security-docs-manual.gradle:19]: Can this be simplified to:

```groovy
def samplesBranch = project.version.replaceAll(/(\d+\.\d+\.).*/, '$1x');
```

File: docs/manual/spring-security-docs-manual.gradle [hunk 2]
```diff
@@ -62,3 +63,12 @@ def resolvedVersions(Configuration configuration) {
 repositories {
 	maven { url "https://repo.spring.io/release" }
 }
+
+def computeSamplesBranch(String version) {
+	if ('main' == version) {
```
Comment (jzheaux) [inline docs/manual/spring-security-docs-manual.gradle:68]: I don't think I understand... when could `project.version` be `main`? I would have thought `project.version` would always be a snapshot, milestone, or release version of the artifact.

File: docs/manual/spring-security-docs-manual.gradle [hunk 3]
```diff
@@ -16,9 +16,10 @@ asciidoctor {
 asciidoctorj {
 	def docsTag = snapshotBuild ? 'current' : project.version
 	def ghTag = snapshotBuild ? 'main' : project.version
+	def samplesBranch = computeSamplesBranch(project.version as String)
 	def ghUrl = "https://github.com/spring-projects/spring-security/tree/$ghTag"
 	def ghOldSamplesUrl = "https://github.com/spring-projects/spring-security/tree/5.4.x/samples"
-	def ghSamplesUrl = "https://github.com/spring-projects/spring-security-samples/tree/$ghTag"
+	def ghSamplesUrl = "https://github.com/spring-projects/spring-security-samples/tree/$samplesBranch"
```
Comment (jzheaux) [inline docs/manual/spring-security-docs-manual.gradle:22]: Should this same change be made in `spring-security-docs-guides`?

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @marcusdacoregio! I've left some feedback inline.

File: docs/manual/spring-security-docs-manual.gradle (hunk 1 above)
Comment (marcusdacoregio) [reply docs/manual/spring-security-docs-manual.gradle:19]: Yeah, and it cover more scenarios, thanks!

File: docs/manual/spring-security-docs-manual.gradle (hunk 2 above)
Comment (marcusdacoregio) [reply docs/manual/spring-security-docs-manual.gradle:68]: I've made a mistake between version and branch. Gonna fix it

**Comment by marcusdacoregio** [comment]
Thanks for your review @jzheaux. I've introduced a new property in `gradle.properties` named `samplesBranch`. This way each Spring Security version should specify which samples' branch it uses and we avoid computing it from the version property, which is more error-prone.

**Comment by marcusdacoregio** [comment]
I've also split this change into two commits so we can cherry-pick it in other branches easier, like #9846

**Review by jzheaux — APPROVED** [review]

**Pull request merged**